packages/store-sdk/src/session/Provider.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { createContext, useMemo } from 'react'
+import type { FC } from 'react'
+
+import { useLocalStorage } from '../storage/useLocalStorage'
+
+export interface Currency {
+  code: string // USD
+  symbol: string // $
+}
+
+export interface User {
+  id: string // user id
+}
+
+interface Session {
+  locale: string // en-US
+  currency: Currency
+  country: string // BRA
+  channel: string | null
+  region: string | null
+  priceTable: string | null
+  postalCode: string | null
+  user: User | null
+}
+
+export interface ContextValue extends Session {
+  setSession: (session: Session) => void
+}
+
+export const Context = createContext<ContextValue | undefined>(undefined)
```

Comment:
React's createContext api blocks the main thread for quite some time when a default value is passed to it. Since it's already on the global script evaluation, we need to minimize this. The only way I know of minimizing this bocking time is passing undefined to it so react internals don't kick in right away and wait until being used